File: backend/src/main/java/com/gpuoj/controller/AuthController.java
package com.gpuoj.controller;

import com.gpuoj.dto.AuthResponse;
import com.gpuoj.dto.LoginRequest;
import com.gpuoj.dto.RegisterRequest;
import com.gpuoj.service.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/register")
    public Mono<AuthResponse> register(@RequestBody RegisterRequest req) {
        return authService.register(req);
    }

    @PostMapping("/login")
    public Mono<AuthResponse> login(@RequestBody LoginRequest req) {
        return authService.login(req);
    }
}

File: backend/src/main/kotlin/com/gpuoj/controller/AuthController.kt
package com.gpuoj.controller

import com.gpuoj.dto.AuthResponse
import com.gpuoj.dto.LoginRequest
import com.gpuoj.dto.RegisterRequest
import com.gpuoj.service.AuthService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {

    @PostMapping("/register")
    fun register(@RequestBody req: RegisterRequest): Mono<AuthResponse> = authService.register(req)

    @PostMapping("/login")
    fun login(@RequestBody req: LoginRequest): Mono<AuthResponse> = authService.login(req)
}
